### PR TITLE
ASAP-45 Add working groups to tag search

### PR DESCRIPTION
**Pull request by gabiayako**

### CODE REVIEW

File: apps/crn-server/serverless.ts [hunk 1]
```diff
@@ -757,6 +757,26 @@ const serverlessConfig: AWS = {
...
+    algoliaIndexWorkingGroups: {
+      handler:
+        './src/handlers/working-group/algolia-index-working-group-handler.handler',
+      events: [
+        {
+          eventBridge: {
+            eventBus: 'asap-events-${self:provider.stage}',
+            pattern: {
+              source: [eventBusSourceContentful],
+              'detail-type': ['WorkingGroupsPublished'],
```
Comment (marianasrv) [inline apps/crn-server/serverless.ts:769]: Why only on publish?

File: apps/crn-server/src/handlers/working-group/algolia-index-working-group-handler.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  (
+    workingGroupController: WorkingGroupController,
+    algoliaClient: AlgoliaClient<'crn'>,
+  ): EventBridgeHandler<WorkingGroupEvent, WorkingGroupPayload> =>
+  async (event) => {
+    logger.debug(`Event ${event['detail-type']}`);
+    const workingGroupId = event.detail.resourceId;
+
+    try {
+      const workingGroup = await workingGroupController.fetchById(
```
Comment (Riccardo-Brown) [inline apps/crn-server/src/handlers/working-group/algolia-index-working-group-handler.ts:28]: Should we not use the dataprovider directly not via the controller?

**Review by Riccardo-Brown — CHANGES_REQUESTED** [review]
I think we should be using the detail-type `WorkingGroupsPublished` or `WorkingGroupsUnpublished` to determine if the handler should add or remove data from the index, instead of using the result of fetching the record

**Comment by gabiayako** [comment]
@Riccardo-Brown I've followed the pattern in the other webhooks fetching the content using the controller instead of the data provider 

- https://github.com/yldio/asap-hub/blob/master/apps/crn-server/src/handlers/event/algolia-index-event-handler.ts#L24 
- https://github.com/yldio/asap-hub/blob/master/apps/crn-server/src/handlers/external-author/algolia-index-external-author-handler.ts#L22
- https://github.com/yldio/asap-hub/blob/master/apps/crn-server/src/handlers/user/algolia-index-user-handler.ts#L26

About using `WorkingGroupsPublished` or `WorkingGroupsUnpublished` as we're already fetching the record to get the parsed content we will save on algolia, I believe it's consistent to check it the content exists or not to determine if we will save it or remove it. If we add `[Entity]Delete` this would work as well. Also it's consistent with the existing handlers.

**Review by marianasrv — APPROVED** [review]

**Review by Riccardo-Brown — APPROVED** [review]

**Pull request merged**